Cinema-Reservation-System-Front-End-Client-Part-Internship-project/src/app/authentication/shared/authentication.service.ts

```diff
@@ -0,0 +1,62 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { CreateUser } from './create-user.model';
+import { environment } from '../../../environments/environment';
+import { LoginUser } from './login-user.model';
+import { BehaviorSubject, tap } from 'rxjs';
+import { User } from './user-model';
+import { jwtDecode } from 'jwt-decode';
+import { Router } from '@angular/router';
+import { ToastService } from '../../toasts/toast.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthenticationService {
+  private apiUrl = environment.apiUrl;
```

Comment:
Instead of composing the request url in each service from apiUrl and the route itself, create a global interceptor setting the request's  url to the ${environment.apiUrl}/${req.url} where req is a first parametr of the interceptor